artifactory/commands/flexpack/maven.go

```diff
@@ -144,63 +180,159 @@ func setMavenBuildPropertiesOnArtifacts(workingDir, buildName, buildNumber strin
+		// Parse the modified time
+		modTime, err := time.Parse("2006-01-02T15:04:05.999Z", item.Modified)
+		if err != nil {
+			log.Debug("Could not parse modified time for " + item.Name + ": " + err.Error())
+			continue
+		}
+
+		// Only include artifacts modified after cutoff
+		if modTime.After(cutoffTime) {
+			recentArtifacts = append(recentArtifacts, *item)
+			log.Debug(fmt.Sprintf("Including recently deployed artifact: %s (modified: %s)", item.Name, item.Modified))
+		}
+	}
+
+	if len(recentArtifacts) == 0 {
+		log.Warn("No recently deployed artifacts found")
+		return nil
+	}
+
 	// Create build properties in the same format as NPM/traditional implementations
 	timestamp := strconv.FormatInt(time.Now().UnixNano()/int64(time.Millisecond), 10) // Unix milliseconds like NPM
 	buildProps := fmt.Sprintf("build.name=%s;build.number=%s;build.timestamp=%s", buildName, buildNumber, timestamp)
 	if projectKey := buildArgs.GetProject(); projectKey != "" {
 		buildProps += fmt.Sprintf(";build.project=%s", projectKey)
 	}
 
-	// Set build properties on found artifacts
-	propsParams := services.PropsParams{
-		Reader: searchReader,
-		Props:  buildProps,
-	}
+	// Set properties on each recent artifact individually
+	for _, artifact := range recentArtifacts {
+		// ResultItem has Repo, Path, and Name fields already separated
+		// Use AQL to find the exact artifact
+		aqlPattern := fmt.Sprintf(`{"repo":"%s","path":"%s","name":"%s"}`,
+			targetRepo, artifact.Path, artifact.Name)
+
+		log.Debug(fmt.Sprintf("Setting properties on: repo=%s, path=%s, name=%s", targetRepo, artifact.Path, artifact.Name))
+
+		searchParams := services.SearchParams{
+			CommonParams: &specutils.CommonParams{
+				Aql: specutils.Aql{
+					ItemsFind: aqlPattern,
+				},
+			},
+		}
 
-	_, err = servicesManager.SetProps(propsParams)
-	if err != nil {
-		return fmt.Errorf("failed to set build properties on artifacts: %w", err)
+		reader, err := servicesManager.SearchFiles(searchParams)
+		if err != nil {
+			log.Warn(fmt.Sprintf("Failed to search for artifact %s: %s", artifact.Name, err))
+			continue
+		}
+
+		propsParams := services.PropsParams{
+			Reader: reader,
+			Props:  buildProps,
+		}
+
+		_, err = servicesManager.SetProps(propsParams)
+		reader.Close()
+		if err != nil {
+			log.Warn(fmt.Sprintf("Failed to set properties on %s: %s", artifact.Name, err))
+		} else {
+			log.Debug("Successfully set properties on: " + artifact.Name)
+		}
 	}
 
 	log.Info("Successfully set build properties on deployed Maven artifacts")
 	return nil
 }
 
-// getMavenArtifactCoordinates extracts Maven coordinates from pom.xml
-func getMavenArtifactCoordinates(workingDir string) (groupId, artifactId, version string, err error) {
-	// Read pom.xml to get project information
+// getMavenDeployRepository determines where Maven deployed artifacts
+// by parsing pom.xml distributionManagement
+func getMavenDeployRepository(workingDir string) (string, error) {
```

Comment:
what if the the deployment repository is mentioned in settings.xml? is it handled?